src/bridge/memory_util/openssl_structs_test.cc

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <openssl/rsa.h>
+
+#include "src/bridge/memory_util/openssl_structs.h"
+
+namespace kmsengine {
+namespace bridge {
+namespace memory_util {
+namespace {
+
+TEST(OpenSSLMakeTest, MakeRsaSetsDeleter) {
+  auto rsa = MakeRsa();
+  EXPECT_NE(rsa.get(), nullptr);
+  EXPECT_EQ(rsa.get_deleter(), &RSA_free);
+}
+
+TEST(OpenSSLMakeTest, MakeRsaMethodSetsDeleter) {
+  auto rsa_method = MakeRsaMethod("", 0);
+  EXPECT_NE(rsa_method.get(), nullptr);
+  EXPECT_EQ(rsa_method.get_deleter(), &RSA_meth_free);
+}
+
+TEST(OpenSSLMakeTest, MakeRsaMethodSetsName) {
+  auto empty_name = MakeRsaMethod("", 0);
+  ASSERT_NE(empty_name.get(), nullptr);
```

Comment:
Why `ASSERT_NE` instead of `EXPECT_NE` (I think the expect would give you a better error)?

Reply:
The subsequent call to `RSA_meth_get0_name` may crash if the input pointer is null.